File: Data Structures & Algorithms/same-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        queue<TreeNode*> pq;
        queue<TreeNode*> qq;
        pq.push(p);
        qq.push(q);

        while (!pq.empty() && !qq.empty()) {
            TreeNode* np = pq.front();
            pq.pop();
            TreeNode* nq = qq.front();
            qq.pop();

            if (!np && !nq) {
                continue;
            }
            if (!np || !nq) {
                return false;
            }
            if (np->val != nq->val) {
                return false;
            }
            pq.push(np->left);
            pq.push(np->right);
            qq.push(nq->left);
            qq.push(nq->right);
        }

        return pq.size() == qq.size();
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        view = []
        queue = deque()
        if root is not None:
            queue.append(root)
        while len(queue) > 0:
            level_length = len(queue)
            view.append(queue[0].val)
            for _ in range(level_length):
                node = queue.popleft()
                if node.right is not None:
                    queue.append(node.right)
                if node.left is not None:
                    queue.append(node.left)
        
        return view